invite: Migrate invite modal banners to the new banner component.

sayamsamal opened the pull request:
This PR migrates the invite modal banners to the new banner component system to fix the issues with the drop shadows and, inconsistent margins and colors.

Fixes: [#issues > URL display in invite modal](https://chat.zulip.org/#narrow/channel/9-issues/topic/URL.20display.20in.20invite.20modal/with/2284987)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
| Before | After |
|--------|--------|
| <img width="708" height="339" alt="Screenshot 2025-11-05 at 3 56 20 AM" src="https://github.com/user-attachments/assets/813142c0-6f3e-4172-a1aa-82b3f309df50" /> | <img width="708" height="339" alt="Screenshot 2025-11-05 at 3 55 03 AM" src="https://github.com/user-attachments/assets/0220a5f3-5b1c-4cae-b92a-17f9b305e8b9" /> |
| <img width="708" height="339" alt="Screenshot 2025-11-05 at 3 56 08 AM" src="https://github.com/user-attachments/assets/9039c64d-9294-424d-8f3c-c8ea1e0434ee" /> | <img width="708" height="339" alt="Screenshot 2025-11-05 at 3 54 45 AM" src="https://github.com/user-attachments/assets/bf57eebf-02ee-4b4e-8a31-b55e14403a53" /> |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
@sahil839 Are you up for reviewing this one? I haven't tested yet.

sahil839 commented:
Looks good to me. @alya do you want to test this once or should this be marked for integration review?

alya commented:
The banner isn't getting cleared properly when you've done all the steps:

<img width="1190" height="314" alt="Screenshot 2025-11-05 at 16 08 42@2x" src="https://github.com/user-attachments/assets/71c1579c-ec61-4b4c-bd48-0879dde00e34" />

alya commented:
When the "default user settings" step is complete, would it be possible to remove that part of the instruction?

When I changed a default user setting, I still saw the full instruction, not just the part about custom profile fileds.

When I went in the opposite direction and just added a custom profile field, the whole instruction disappeared -- maybe we decided to do that intentionally?

sayamsamal commented:
> The banner isn't getting cleared properly when you've done all the steps:
> 
> <img alt="Screenshot 2025-11-05 at 16 08 42@2x" width="1190" height="314" src="https://private-user-images.githubusercontent.com/2090066/510472164-71c1579c-ec61-4b4c-bd48-0879dde00e34.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI1MTc4NTYsIm5iZiI6MTc2MjUxNzU1NiwicGF0aCI6Ii8yMDkwMDY2LzUxMDQ3MjE2NC03MWMxNTc5Yy1lYzYxLTRiNGMtYmQ0OC0wODc5ZGRlMDBlMzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTEwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMDdUMTIxMjM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWVlOWZlMDY0YmFjZmY1NjA0YTFmNDQ5OWRmZjc4ZDFkZWI1YjhjMmNjZjVkZDA0ZjE5ZTMwOWQ5M2Q0YTE0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.52C00zjHkdfpAMGPZiAIL6V0AZwsZUbSMdAiPpaWkIU">

Good catch! And fixed it in the recent push.

sayamsamal commented:
> When the "default user settings" step is complete, would it be possible to remove that part of the instruction?
> 
> When I changed a default user setting, I still saw the full instruction, not just the part about custom profile fileds.
> 
> When I went in the opposite direction and just added a custom profile field, the whole instruction disappeared -- maybe we decided to do that intentionally?

Yep, that was decided in [#design > setup tips in user invite modal @ 💬](https://chat.zulip.org/#narrow/channel/101-design/topic/setup.20tips.20in.20user.20invite.20modal/near/1684373). We basically didn't have a proper method to verify if the default user settings have been changed.

timabbott commented on web/src/invite.ts:
```diff
@@ -226,7 +234,7 @@ function submit_invitation_form(): void {
             if (!parsed.success) {
                 // There was a fatal error, no partial processing occurred.
-                ui_report.error("", xhr, $invite_status);
+                ui_report.error($t({defaultMessage: "Failed"}), xhr, $invite_status);
```
I guess is there a future TODO to make `$invite_status` use a banner component as well? Just noticed we replaced the success code path, but not the error code path.

timabbott commented:
Merged, thanks @sayamsamal and @sahil839!

@sahil839 can you do an extra round of manual testing, just in case, since this is an important part of the onboarding experience?

sahil839 commented:
I cannot see tabs for email and multiuse invites when everything is configured and no banners are needed to be shown - 
<img width="848" height="702" alt="image" src="https://github.com/user-attachments/assets/3834c860-0d93-4247-ae67-83de53b7d34b" />

sayamsamal commented:
> I cannot see tabs for email and multiuse invites when everything is configured and no banners are needed to be shown - 

Thanks for catching this bug! And my bad for not noticing the fault in logic, opened #36549 to solve this bug.
